### PR TITLE
FRONT-805: Functional room list and creation

### PR DESCRIPTION
**Pull request by bearni95**
Uses the `client.listStreams` method to fetch the streams the Metamask address is subscribed to. 

It then processes them using the  `ChatRoomManager` lib and assigns them to visual items on the left bar of the chat.

The "Add new room" button at the top of the left bar works for creating rooms with arbitrary-given uuids as names.

### CODE REVIEW

File: src/lib/ChatRoom.ts [hunk 1]
```diff
@@ -1,198 +1,16 @@
-import { StreamrClient, Stream, StreamOperation } from 'streamr-client'
+import { Stream } from 'streamr-client'
 
 export interface ChatMessage {
     type: 'text' | 'metadata'
```
Comment (mondoreale) [inline src/lib/ChatRoom.ts:4]: We may want a new type here – `MessageType` (or `ChatMessageType`).

```ts
enum MessageType {
    Text = 'text',
    Metadata = 'metadata',
}

export interface ChatMessage {
    type: MessageType
    payload: string
}
```

File: src/lib/ChatRoomManager.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    public async publishMessage(
+        streamId: string,
+        message: string
+    ): Promise<any> {
+        return this.client.publish(
+            streamId,
+            {
+                type: 'text',
```
Comment (mondoreale) [inline src/lib/ChatRoomManager.ts:152]: `type: MessageType.Text,` from above suggestion maybe?

File: src/lib/ChatRoomManager.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    public async publishMetadata(
+        streamId: string,
+        metadata: any
+    ): Promise<any> {
+        return this.client.publish(
+            streamId,
+            {
+                type: 'metadata',
```
Comment (mondoreale) [inline src/lib/ChatRoomManager.ts:168]: `type: MessageType.Metadata,`

**Review by mondoreale — COMMENTED** [review]
I see what's going on. Let's try to shape the app more like a monolith, style-wise. I left a couple of possibly direction-establishing comments.

Don't hesitate to disagree, please!

File: src/lib/ChatRoomManager.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    public async getInvitations(): Promise<Array<Stream>> {
+        const streams = await this.client.listStreams({
+            operation: StreamOperation.STREAM_SHARE,
+        })
+
+        const address = await this.client.getAddress()
+        const invitations: Array<Stream> = []
+        for (let i = 0; i < streams.length; i++) {
```
[the remark this answers is not in the document]
Comment (bearni95) [reply src/lib/ChatRoomManager.ts]: Nothing against `forEach`, just more used to good, olde `for` loops. Should that change?

**Review by mondoreale — APPROVED** [review]
This is going in the right direction, kind sir.

I like that we got the functional stuff in. There's a lot more we could do to make it more "reacty". One thing at a time.

File: src/components/pages/Chat/AddRoomItem.tsx [hunk 5]
```diff
@@ -1,34 +1,37 @@
...
+            session.wallet!.address,
+            metamaskAddress,
+            id,
+            rooms.map((r) => r.id)
+        )
+        dispatch({
+            type: ActionType.AddRooms,
+            payload: [room],
+        })
+    }, [metamaskAddress, session, rooms])
```
Comment (mondoreale) [inline src/components/pages/Chat/AddRoomItem.tsx:26]: `dispatch` missing.

**Pull request merged**